### DISCUSSION TITLE
@quasar/extras/fontawesome-v6 module error

### DISCUSSION THREAD

**Discussion by ecotrissy** [quasarframework/quasar#13581] [General - Components / Directives / etc] [answered]
I’ve been unable to update quasar correctly for some time.
After updating, I get the following error on build:

```
App •  ERROR  •  UI  in ./node_modules/quasar/icon-set/svg-fontawesome-v6.mjs
Module not found: Can't resolve imported dependency "@quasar/extras/fontawesome-v6"
Did you forget to install it? You can run: yarn add @quasar/extras/fontawesome-v6
```

Running yarn add @quasar/extras/fontawesome-v6 gives me:
```
>yarn add @quasar/extras/fontawesome-v6
yarn add v1.22.15
[1/5] Validating package.json...
[2/5] Resolving packages...
error An unexpected error occurred: "https://registry.yarnpkg.com/@quasar%2fextras/fontawesome-v6: Request \"https://registry.yarnpkg.com/@quasar%2fextras/fontawesome-v6\" returned a 405".
```

**Comment by ecotrissy** [comment]
I'm still suffering the above yarn 405 problem.

I've found what looks like a typo ( **fort**awesome ) in:
https://cdn.jsdelivr.net/npm/@quasar/extras/package.json

```
{
  "name": "@quasar/extras",
  "version": "1.14.0",
  "description": "Quasar Framework fonts, icons and animations",
  "scripts": {
    "build": "node build/index.js"
  },
  "main": "index.js",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/quasarframework/quasar.git"
  },
  "keywords": [
    "quasar",
    "framework",
    "fonts",
    "icons",
    "animations"
  ],
  "author": "Razvan Stoenescu",
  "contributors": [
    {
      "name": "Jeff Galbraith",
      "email": "jeff@quasar.dev",
      "url": "https://jeffgalbraith.dev"
    }
  ],"license": "MIT",
  "bugs": {
    "url": "https://github.com/quasarframework/quasar/issues"
  },
  "funding": {
    "type": "github",
    "url": "https://donate.quasar.dev"
  },
  "homepage": "https://quasar.dev",
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "@fortawesome/fontawesome-free": "6.1.1",
    "@mdi/font": "6.6.96",
    "@mdi/svg": "6.6.96",
    "animate.css": "4.1.1",
    "bootstrap-icons": "^1.8.1",
    "cross-fetch": "^3.1.5",
    "cross-spawn": "7.0.3",
    "eva-icons": "1.1.3",
    "fs-extra": "10.1.0",
    "glob": "8.0.1",
    "ionicons": "6.0.1",
    "line-awesome": "1.3.0",
    "themify-icons": "https://github.com/lykmapipo/themify-icons",
    "@xmldom/xmldom": "0.8.2"
  }
}

```

**Reply by yusufkandemir** [reply]
**fort**awesome is more like the organization behind Font Awesome. So, it's not a typo.
https://github.com/FortAwesome/Font-Awesome

The problem is not related to that, I explained it below(see https://github.com/quasarframework/quasar/discussions/13581#discussioncomment-2890453).

**Comment by yusufkandemir** [comment, marked as answer]
It's not very clear which flavor or versions of Quasar you are using nor how you upgraded what. You are probably using Quasar CLI with Webpack judging from your saying "for some time" and the way resolve errors are structured. Your dependencies might be got messed up somehow, you might have accidentally deleted `@quasar/extras`. You might stay stuck on an older version of `@quasar/extras`, or you might still be on `@quasar/app` instead of `@quasar/app-webpack`, etc. Sharing the output of `quasar info` helps. Make sure to use `quasar upgrade -i` when upgrading your Quasar-related packages instead of doing it manually. Make sure you have the latest version of `@quasar/cli`, `npm i -g @quasar/cli@latest` will install the latest one available(_if you installed it using npm_).

Aside from all these, although the CLI recommends using `yarn add @quasar/extras/fontawesome-v6`, it's invalid. It is just a rough guess from the CLI. If I am not mistaken, any package that has more than 2 levels is wrong. It would either be something like `yarn add whatever` or `yarn add @org/whatever`, no deeper than that. So, you need to do `yarn add @quasar/extras` instead, it will install it if you accidentally deleted it before.

**Reply by ecotrissy** [reply]
Thank you very much for this.
I think the problem was that I was on @quasar/app instead of @quasar/app-webpack, as you suggested.
All good now!
